cmds/core/head/head.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2024 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+var errCombine = fmt.Errorf("can't combine line and byte counts")
+
+func run(stdout io.Writer, stderr io.Writer, bytes, count int, files ...string) error {
+	if bytes > 0 && count > 0 {
+		return errCombine
+	}
+
+	var printBytes bool
+	var buffer []byte
+	if bytes > 0 {
+		printBytes = true
+		buffer = make([]byte, bytes)
+	}
+
+	if count == 0 {
+		count = 10
+	}
+
+	var newLineHeader bool
+
+	for _, file := range files {
+		f, err := os.Open(file)
+		if err != nil {
+			fmt.Fprintf(stderr, "head: %v", err)
```

Comment:
if you were to use errors.Join, then a user could see a list of all the file names that failed at the end of the command, where they would be much more visible than if they were interspersed with output? Then we'd have no need to write to stderr in this function.

Reply:
All head implementations that I know (not many to be honest) prints errors in stderr in order of [FILES] and don't wait until the end. So I think it will be better to stay consistent.

Reply:
it's one of the things that most annoys me about head, tbh, so, pretty please, can we show the failures at the end? It would be a real improvement. Your call of course.